core/esmf-aspect-meta-model-java/src/test/java/org/eclipse/esmf/aspectmodel/loader/AspectModelLoaderTest.java

```diff
@@ -29,18 +37,76 @@
+
+      assertThat( aspectModel.namespaces() ).hasSize( 2 );
+      assertThat( aspectModel.namespaces().get( 0 ).getName() ).contains( "urn:samm:org.eclipse.examples:1.1.0" );
+      assertThat( aspectModel.namespaces().get( 1 ).getName() ).contains( "urn:samm:org.eclipse.examples:1.0.0" );
+
+      final List<String> aspectsNames = List.of( "Movement2", "Movement3", "Movement", "SimpleAspect" );
+
+      assertThat( aspectModel.files() ).hasSize( 4 );
+      assertThat( aspectModel.files() )
+            .anySatisfy( aspectModelFile -> {
+               assertThat( aspectsNames ).contains( aspectModelFile.aspect().getName() );
+            } );
+   }
+
+   @Test
+   void testLoadAspectModelFromZipArchiveInputStream() throws URISyntaxException, FileNotFoundException {
+      final ClassLoader classLoader = getClass().getClassLoader();
+      final Path archivePath = Paths.get( classLoader.getResource( "namespaces.zip" ).toURI() );
+      final AspectModel aspectModel = new AspectModelLoader().loadNamespacePackage( new FileInputStream( archivePath.toString() ) );
+
+      assertThat( aspectModel.namespaces() ).hasSize( 2 );
+      assertThat( aspectModel.namespaces().get( 0 ).getName() ).contains( "urn:samm:org.eclipse.examples:1.1.0" );
+      assertThat( aspectModel.namespaces().get( 1 ).getName() ).contains( "urn:samm:org.eclipse.examples:1.0.0" );
+
+      final List<String> aspectsNames = List.of( "Movement2", "Movement3", "Movement", "SimpleAspect" );
+
+      assertThat( aspectModel.files() ).hasSize( 4 );
+      assertThat( aspectModel.files() )
+            .anySatisfy( aspectModelFile -> {
+               assertThat( aspectsNames ).contains( aspectModelFile.aspect().getName() );
+            } );
+   }
+
+   /**
+    * Test migration to the latest version of Aspect Model in Archive
+    */
+   @Test
+   void testLoadAspectModelFromZipArchive2_0_0() throws URISyntaxException, FileNotFoundException {
+      final ClassLoader classLoader = getClass().getClassLoader();
+      final Path archivePath = Paths.get( classLoader.getResource( "namespaces_with_old_version.zip" ).toURI() );
+      final AspectModel aspectModel = new AspectModelLoader().loadNamespacePackage( new FileInputStream( archivePath.toString() ) );
+
+      assertThat( aspectModel.namespaces() ).hasSize( 2 );
+      assertThat( aspectModel.namespaces().get( 0 ).getName() ).contains( "urn:samm:org.eclipse.examples:1.1.0" );
+      assertThat( aspectModel.namespaces().get( 1 ).getName() ).contains( "urn:samm:org.eclipse.examples:1.0.0" );
+
+      final List<String> aspectsNames = List.of( "Movement2", "Movement3", "Movement4", "Movement", "SimpleAspect" );
+
+      assertThat( aspectModel.files() ).hasSize( 5 );
+      assertThat( aspectModel.files() )
+            .anySatisfy( aspectModelFile -> {
+               assertThat( aspectsNames ).contains( aspectModelFile.aspect().getName() );
+            } );
    }
```

Comment:
Can we add a test here for loading a namespace archive that contains a model that references a URN that is not contained in the archive, but is resolved using the ResolutionStrategy passed to the AspectModelLoader?
This would make this case more robust.